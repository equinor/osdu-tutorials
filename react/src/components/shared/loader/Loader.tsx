import React from 'react';
import './styles.css';

interface Props {
  // a font size. changes size of a componens as height and width are specified in "em"
  size?: number;
}

/**
 * super-duper animated bars will keep you amused
 * while little electrons rushes through cables under an ocean
 */
export function Loader({ size = 24 }: Props) {
  return (
    <div className="loader">
      <div className="loader__component" style={{ fontSize: size }}></div>
    </div>
  );
}
