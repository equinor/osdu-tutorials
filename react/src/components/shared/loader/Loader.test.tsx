import React from 'react';
import { shallow } from 'enzyme';
import { Loader } from './Loader';

describe('Loader', () => {
  it('should render self and subcomponents', () => {
    const wrapper = shallow(<Loader />);

    expect(wrapper).toMatchSnapshot();
  });
});
