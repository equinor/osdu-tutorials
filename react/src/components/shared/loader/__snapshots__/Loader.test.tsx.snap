// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should render self and subcomponents 1`] = `
<div
  className="loader"
>
  <div
    className="loader__component"
    style={
      Object {
        "fontSize": 24,
      }
    }
  />
</div>
`;
